### PR TITLE
Add translator notes to Gettext strings and update makepot task

### LINKED ISSUE #617: Add gettext context for strings
Translations team [requested context](https://godaddy.slack.com/archives/C07TDQWJ1/p1695676984734869?thread_ts=1695360403.201719&cid=C07TDQWJ1) for the following, which could be provided by translator notes or context arg:
	
```
#: admin/abstract-sv-wc-plugin-admin-setup-wizard.php:608
msgid "%s is ready!"
#: class-sv-wc-framework-bootstrap.php:282
msgid ""
"To resolve this, please %1$supdate%2$s (recommended) %3$sor%4$s "
"%5$sdeactivate%6$s the above plugin, or %7$sdeactivate the following%8$s:"
#: class-sv-wc-plugin-dependencies.php:228
msgid "%s or higher"
#: payment-gateway/External_Checkout/apple-pay/class-sv-wc-payment-gateway-apple-pay-frontend.php:176
msgid "Buy with"
#: payment-gateway/External_Checkout/apple-pay/class-sv-wc-payment-gateway-apple-pay-admin.php:101
msgid "Allow Apple Pay on"
#: payment-gateway/External_Checkout/apple-pay/class-sv-wc-payment-gateway-apple-pay-admin.php:116
msgid "White with outline"
#: class-sv-wc-plugin.php:709
msgid "%1$s - A minimum of %2$s is required."
#: class-sv-wc-plugin.php:718
msgid "Set as %1$s - %2$s is required."
#: payment-gateway/class-sv-wc-payment-gateway-privacy.php:209
msgid "%s Payment Tokens"
#: payment-gateway/class-sv-wc-payment-gateway-direct.php:503
msgid "Payment method address could not be updated. %s"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php:892
msgid "%s is required."
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:261
msgid "Pre-Order Tokenization attempt failed (%s)"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:307
msgid "%s - Pre-Order Release Payment for Order %s"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:311
msgid "Payment token missing/invalid."
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:336
msgid "%s %s Pre-Order Release Payment Approved: %s ending in %s (expires %s)"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:347
msgid "%s eCheck Pre-Order Release Payment Approved: %s ending in %s"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:391
msgid "Pre-Order Release Payment Failed: %s"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php:384
msgid "%1$s - Subscription Renewal Order %2$s"
#: payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php:671
msgid "Via %s ending in %s"
#: rest-api/Controllers/Settings.php:224
msgid "Could not update setting: %s"
There are several multi-lines strings. Please provide string in a single line, linguists can adjust the order. Such as
#: rest-api/Controllers/Settings.php:342
msgid ""
"Optional object that defines how the user will interact with and update the "
"setting."
-> "Optional object that defines how the user will interact with and update the setting."
```

### PR DESCRIPTION
**Pull request by unfulvio-godaddy**
Adds some translators notes requested on #617 via [Slack](https://godaddy.slack.com/archives/C07TDQWJ1/p1695676984734869?thread_ts=1695360403.201719&cid=C07TDQWJ1) -- Closes #617 

About this item:

> There are several multi-lines strings. Please provide string in a single line, linguists can adjust the order. Such as
#: rest-api/Controllers/Settings.php:342
msgid ""
"Optional object that defines how the user will interact with and update the "
"setting."
-> "Optional object that defines how the user will interact with and update the setting."
> 

This seems to be the output from our potfile builder, not the source, which truncates lines at 80 chars. I could not find an option in https://github.com/cedaro/grunt-wp-i18n/blob/develop/docs/makepot.md to change this. 

We should probably switch to https://developer.wordpress.org/cli/commands/i18n/make-pot/ (something that is [recommended also on the Grunt project above](https://github.com/cedaro/grunt-wp-i18n/issues/80)) as this other tool does not seem to exhibit the problem. I made a quick test from CLI and runs fine. We could do so with this PR and later update Sake to do the same in plugins. This would require the user to have installed WP CLI (or later have scripts that install the dependency to correctly handle the command).

~~But I could probably reach the translations team and tell to ignore the line breaks.~~ Nope: they want it without line breaks: https://github.com/godaddy-wordpress/wc-plugin-framework/pull/618#issuecomment-1734876985

I have updated the `makepot` command to switch to WP CLI. I wanted to run a local instance of WP CLI loaded via Composer as dev dependency. However, it turns out that since we bundle Codeception one such dependency already exists and I think it ends up running an older version that doesn't include the `i18n` command. So I'm a bit stuck there and opted to use a global WP CLI instance - this means if the person running the script doesn't have WP CLI installed, it will fail.

## QA

- [x] The cases prompted by #617 are addressed
- [x] Try to run `grunt update_translations`: the potfile should be generated via WP CLI without line/breaks

No code change, no tests or changelog needed

### CODE REVIEW

**Comment by unfulvio-godaddy** [comment]
edit: it appears that the GoLF team prefers not having multi-line strings to translate - I can imagine this could become troublesome in their system, especially if those lines change - perhaps we should try to implement the new potfile generator in the same context as this PR

File: woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php [hunk 1]
```diff
@@ -333,18 +338,25 @@ public function process_release_payment( $order ) {
 				if ( $this->get_gateway()->is_credit_card_gateway() ) {
 
 					$message = sprintf(
-							__( '%s %s Pre-Order Release Payment Approved: %s ending in %s (expires %s)', 'woocommerce-plugin-framework' ),
+							/* translators: Context: A payment is released for a pre-order. Placeholders: %1$s - Payment gateway name, %2$s - Either 'Authorization' or 'Charge' (untranslated), %3$s - Payment method type (e.g. 'PayPal', 'Credit Card', etc.), %4$s - Last four digits of the card or account, %5$s - Expiration date of the payment method */
```
Comment (itambek-godaddy) [inline woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:341]: > Either 'Authorization' or 'Charge' (untranslated)

Is there a reason why these are untranslated? Perhaps something that was simply initially missed?

File: woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php [hunk 2]
```diff
@@ -884,11 +896,13 @@ public function admin_validate_payment_meta( $meta ) {
 
 		// payment token
 		if ( empty( $meta['post_meta'][ $prefix . 'payment_token' ]['value'] ) ) {
+			/* translators: Context: Error message. Placeholder: %s - Required value not found or not as expected */
```
Comment (itambek-godaddy) [inline woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php:899]: This seems confusing to me. How about:

```suggestion
			/* translators: Context: Error message. Placeholder: %s - Label of the related value */
```

File: woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php [hunk 3]
```diff
@@ -884,11 +896,13 @@ public function admin_validate_payment_meta( $meta ) {
...
 			throw new \Exception( sprintf( __( '%s is required.', 'woocommerce-plugin-framework' ), $meta['post_meta'][ $prefix . 'payment_token' ]['label'] ) );
 		}
 
 		// customer ID - optional for some gateways so check if it's set first
 		if ( isset( $meta['post_meta'][ $prefix . 'customer_id'] ) && empty( $meta['post_meta'][ $prefix . 'customer_id' ]['value'] ) ) {
+			/* translators: Context: Error message. Placeholder: %s - Required value not found or not as expected */
```
Comment (itambek-godaddy) [inline woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php:905]:
```suggestion
			/* translators: Context: Error message. Placeholder: %s - Label of the related value */
```

Same as abov

File: woocommerce/class-sv-wc-plugin.php [hunk 4]
```diff
@@ -715,6 +716,7 @@ public function add_system_status_php_information( $rows ) {
 					continue;
 				}
 
+				/* translators: Placeholders: %1$s - PHP setting current value, %2$s - required value */
```
Comment (itambek-godaddy) [inline woocommerce/class-sv-wc-plugin.php:719]:
```suggestion
				/* translators: As in "Value has been set as [foo], but [bar] is required". Placeholders: %1$s - current value for a PHP setting, %2$s - required value for teh PHP setting */
```

File: woocommerce/class-sv-wc-plugin.php [hunk 5]
```diff
@@ -706,6 +706,7 @@ public function add_system_status_php_information( $rows ) {
 					continue;
 				}
 
+				/* translators: Placeholders: %1$s - PHP setting value, %2$s - version or value currently set */
```
Comment (itambek-godaddy) [inline woocommerce/class-sv-wc-plugin.php:709]: The message itself suggests a different placeholder context, no?

```suggestion
				/* translators: Placeholders: %1$s - PHP setting value, %2$s - version or value required */
```

File: woocommerce/class-sv-wc-framework-bootstrap.php [hunk 6]
```diff
@@ -278,7 +278,7 @@ public function render_update_notices() {
 				// describe the way to fix it
 				echo '<p>';
 					printf(
-						/** translators: Placeholders: %1$s - <a> tag, %2$s - </a> tag, %3$s - <em> tag, %4$s - </em> tag, %5$s - <a> tag, %6$s - </a> tag, %7$s - <a> tag, %8$s - </a> tag */
+						/* translators: Context is for plugins to activate or deactivate tags. HTML Placeholders: %1$s - <a> tag, %2$s - </a> tag, %3$s - <em> tag, %4$s - </em> tag, %5$s - <a> tag, %6$s - </a> tag, %7$s - <a> tag, %8$s - </a> tag */
```
Comment (itambek-godaddy) [inline woocommerce/class-sv-wc-framework-bootstrap.php:281]: The word `tags` here seems to be a leftover from refactoring, unless I am missing something?

```suggestion
						/* translators: Context is for plugins to activate or deactivate. HTML Placeholders: %1$s - <a> tag, %2$s - </a> tag, %3$s - <em> tag, %4$s - </em> tag, %5$s - <a> tag, %6$s - </a> tag, %7$s - <a> tag, %8$s - </a> tag */
```

**Review by itambek-godaddy — CHANGES_REQUESTED** [review]
Looking good, although I would tweak the comments here and there.

**Comment by itambek-godaddy** [comment]
> We should probably switch to https://developer.wordpress.org/cli/commands/i18n/make-pot/ (something that is [recommended also on the Grunt project above](https://github.com/cedaro/grunt-wp-i18n/issues/80)) as this other tool does not seem to exhibit the problem. I made a quick test from CLI and runs fine. We could do so with this PR and later update Sake to do the same in plugins. This would require the user to have installed WP CLI (or later have scripts that install the dependency to correctly handle the command).

@unfulvio-godaddy with GoLF integration in place, this should already be done for us with GitHub Actions. It looks like the actions to update and send files to translations have not yet been set up in this repo, though. I'm going to try going through the rest of the setup steps today, to hopefully finish the integration.

However, since this is an open-source project, I think your suggestion to switch from `grunt-wp-i18n` to `wp-cli i18n` makes sense (also, for `sake` to use the cli tool).

File: woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php (hunk 1 above)
Comment (unfulvio-godaddy) [reply woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-pre-orders.php:341]: I am not sure unfortunately, and I don't know if there's any legal requirement to keep them as such

File: woocommerce/class-sv-wc-framework-bootstrap.php (hunk 6 above)
Comment (unfulvio-godaddy) [reply woocommerce/class-sv-wc-framework-bootstrap.php:281]: typo on my end

File: woocommerce/class-sv-wc-plugin.php [hunk 7]
```diff
@@ -715,6 +716,7 @@ public function add_system_status_php_information( $rows ) {
 					continue;
 				}
 
+				/* translators: As in "Value has been set as [foo], but [bar] is required". Placeholders: %1$s - current value for a PHP setting, %2$s - required value for teh PHP setting */
```
Comment (itambek-godaddy) [inline woocommerce/class-sv-wc-plugin.php:719]:
```suggestion
				/* translators: As in "Value has been set as [foo], but [bar] is required". Placeholders: %1$s - current value for a PHP setting, %2$s - required value for the PHP setting */
```

**Review by itambek-godaddy — APPROVED** [review]
LGTM

**Pull request merged**